File: src/main/java/me/kuku/yuq/dao/GroupDao.java
package me.kuku.yuq.dao;

import com.icecreamqaq.yudb.YuDao;
import com.icecreamqaq.yudb.jpa.annotation.Dao;
import com.icecreamqaq.yudb.jpa.annotation.Select;
import me.kuku.yuq.entity.GroupEntity;

import java.util.List;

@Dao
public interface GroupDao extends YuDao<GroupEntity, Integer> {
    GroupEntity findByGroup(Long group);
    List<GroupEntity> findByOnTimeAlarm(Boolean onTimeAlarm);
    @Select("from GroupEntity")
    List<GroupEntity> findAll();
    List<GroupEntity> findByLocMonitor(Boolean locMonitor);
}

File: src/main/kotlin/me/kuku/yuq/dao/GroupDao.kt
package me.kuku.yuq.dao

import com.icecreamqaq.yudb.YuDao
import com.icecreamqaq.yudb.jpa.annotation.Dao
import com.icecreamqaq.yudb.jpa.annotation.Select
import me.kuku.yuq.entity.GroupEntity

@Dao
interface GroupDao: YuDao<GroupEntity, Int>{
    fun findByGroup(group: Long): GroupEntity?
    fun findByOnTimeAlarm(onTimeAlarm: Boolean): List<GroupEntity>
    @Select("from GroupEntity")
    fun findAll(): List<GroupEntity>
    fun findByLocMonitor(locMonitor: Boolean): List<GroupEntity>
}